src/NewTools-Sindarin-Tools/StSindarinDebuggerPresenter.class.st

```diff
@@ -11,6 +11,12 @@ Class {
 	#package : 'NewTools-Sindarin-Tools'
 }
 
+{ #category : 'accessing' }
```

Comment:
When you inherit from StPresenter you get automatically the application so you do not need currentApplication
just self application should return it.

Reply:
> When you inherit from StPresenter you get automatically the application so you do not need currentApplication just self application should return it.

Actually it inherits from SpPresenter, which returns SpNullApplication instead of StPharoApplication.
It inherits from it to inherit from StInspectionPresenter.
Looking at the presenter, I don't see the reason why, so it could be migrated to StPresenter hierarchy instead.

Reply:
It's probably historical. IIRC Sindarin started before NewTools was in place

I'd be great to have it inherit from StPresenter if possible

Reply:
I agree, I will see to that tomorrow. Should be simple.